### ISSUE TITLE
PSA: lens_calibrate.py produced vignetting calibrations are invalid

### ISSUE THREAD

**Issue by LebedevRI** [lensfun/lensfun#494]
All the vignetting calibrations that were made with that new script should be purged.
@cryptomilk @junkyardsparkle 
(FIXME: and probably not only vignetting?)

The script uses darktable to convert raws,
which is a **very** good step in the right direction.
I can't stress that enough.
But it does it wrongly.

You want to produce the model (vignetting, in this case),
on the image in the same 'point' of the processing pipeline,
where said model will be used for correction.
In all sane cases it is in camera native color space.

In darktable's case: (standard history stack)
![image](https://user-images.githubusercontent.com/88600/50789337-fdab4880-12cc-11e9-8f12-f954052f3c98.png)
you want to disable every module after `lens correction` module.

The script kind-of does that:
https://gitlab.com/cryptomilk/lens_calibrate/blob/665dbc7cb92bb98847a640760ebd625fb1d041cd/lens_calibrate.py#L299-313
https://gitlab.com/cryptomilk/lens_calibrate/blob/665dbc7cb92bb98847a640760ebd625fb1d041cd/lens_calibrate.py#L157-187
It disables `sharpen`, `orientation`, `basecurve`, sets `output color profile` to the correct `linear rec2020`.
BUT the `input color profile` is still unchanged...
It should too be set to `linear rec2020`.
That way, since both the input and output color profile is the same,
it acts as a passthrough, and keeps the camera color space.

**Comment by junkyardsparkle** [comment]
I'm not personally aware of any such profiles having been commited (I'm still using the old dcraw script myself) but if anyone knows otherwise, speak up. Hopefully anybody doing vignetting calibration with that script would have noticed the bad results upon testing, as the person reporting on the LensFun list did. This is one reason that I'm a little but uncomfortable with adding submitted vignetting data to the repository... without the lens in hand for testing, you can't really sanity-check the data. :/

> (FIXME: and probably not only vignetting?)

TCA correction could theoretically be messed up with crazy enough color shifts, but I'm pretty sure just the input color profile wouldn't cause a problem, as long as red, green, and blue still resemble red, green, and blue... and distortion correction doesn't care about color at all, obviously.

I've wondered if some of the other software using the profiles (particularly the proprietary Windows type) gets this right in their processing pipeline; it would be an easy bug for them to miss, given that only a minority of profiles include vignetting correction. It may also be an issue with the Gimp plug-in.

**Comment by sarunasb** [comment]
On 1/7/19 3:39 PM, junkyardsparkle wrote:
> I'm not personally aware of any such profiles having been commited (I'm
> still using the old dcraw script myself) but if anyone knows otherwise,
> speak up. 

I haven't used the new script yet.

-- 
Šarūnas Burdulis
math.dartmouth.edu/~sarunas

**Comment by cryptomilk** [comment]
Could you please check:

https://gitlab.com/cryptomilk/lens_calibrate/commit/1d07071510a5320e6ac428b39bf452c54f67fa1a

**Comment by cryptomilk** [comment]
Btw. can someone explain the created plots and how to interpret them?

**Comment by sarunasb** [comment]
For the calibration upload (e86eed #489), which I pushed yesterday to lensfun I also just tried the new script (adjusted k values to 4 decimal difits with round()). Comparing k values for vignetting correction, they are all different, but by not by much, sometimes 4th digit only. I also did a darktable's color picker test for images corrected with values from both old and new scripts (use color picker in Lab, compare L values for center and corners). There are differences, but non-essential, IMO. Visually corrected images are fine. If interested, here are L values measured for 10 images from the calibration set: https://math.dartmouth.edu/nextcloud/s/qjaEtEXaMWBj98Z
